File: Assignment_No19/File4.c
#include<stdio.h>

/*
4 4 4 4 4
3 3 3 3 3
2 2 2 2 2
1 1 1 1 1
*/
void Pattern(int iRows, int iCol)
{
    int i=0, j=0;

    for(i=iRows; i>0; i--)
    {
        for(j=1; j<=iCol; j++)
        {
            printf("%d\t", i);
        }
            printf("\n");
    }
}
int main()
{
    int iValue1=0 , iValue2=0;

    printf("Enter the number of rows and columns\n");
    scanf("%d %d",&iValue1,&iValue2);

    Pattern(iValue1,iValue2);

    return 0;
}

File: Assignment_No47/File4.java
import java.util.*;
import LB.Matrix;

class File4
{
    public static void main(String[] arg)
    {
        boolean bRet = false;
        int iRow = 0, iCol = 0;
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of rows:");
        iRow = sobj.nextInt();

        System.out.println("Enter number of columns:");
        iCol = sobj.nextInt();

        MyMatrix mobj = new MyMatrix(iRow,iCol);

        mobj.Accept();

        mobj.Display();

        bRet =  mobj.ChkIdentity();

        if(bRet)
        {
            System.out.println("The given matrix is identity matrix.\n");
        }
        else
        {
            System.out.println("The given matrix is not an identity matrix.\n");
        }

    }
}

class MyMatrix extends Matrix
{
    public MyMatrix(int Rows, int Cols)
    {
        super(Rows,Cols);
    }

    public boolean ChkIdentity()
    {
        boolean bFlag = true;

        for(int i = 0; i < Arr.length; i++)
        {
            for(int j = 0; j < Arr[i].length; j++)
            {
                if((i==j) && (Arr[i][j] != 1))
                {
                    bFlag = false;
                    break;
                }
                else if((i !=j) && Arr[i][j] != 0)
                {
                    bFlag = false;
                    break;
                }
            }
        }
        return bFlag;
    }
} 